jmetal-core/src/main/java/org/uma/jmetal/measure/impl/CountingMeasure.java

```diff
@@ -181,21 +181,32 @@ public void measureGenerated(T value) {
 			linkedMeasures.remove(measure);
 		}
 	}
+	
+	/*
+	 * (non-Javadoc)
+	 * The Object.finalize() method is called on an object by the garbage collector 
+	 * when it determines that there are no more references to the object. But there 
+	 * is absolutely no warranty that this method will be called AS SOON AS the last 
+	 * references to the object are removed. It can be few microseconds to few minutes 
+	 * later. So when system resources need to be disposed by an object, it's better 
+	 * to not rely on this asynchronous mechanism to dispose them.
+	 */
 
-	@Override
+/*	@Override
 	protected void finalize() throws Throwable {
-		/*
+		*
 		 * Use an intermediary collection for the loop to avoid the concurrent
 		 * accesses leading to exceptions or inconsistent states.
-		 */
+		 *
 		Collection<PushMeasure<?>> remainingMeasures = new LinkedList<>(
 				linkedMeasures.keySet());
 		for (PushMeasure<?> measure : remainingMeasures) {
 			unlink(measure);
 		}
 		super.finalize();
 	}
-
+*/
+	
```

Comment:
This finalize is for ensuring safety: if someone forget to call `unlink()`, the finalizer do it. If it is used correctly, the finalizer wont have anything to do.

Removing a safety for the sole reason of expecting a perfect use is dangerous.